core/modules/layout/layout.flexible.inc

```diff
@@ -0,0 +1,1167 @@
+  return $form['row_settings'];
+}
+
+/**
+ * AJAX responder on the submit button for
+ * layout_flexible_template_edit_row_form().
+ */
+function layout_flexible_template_edit_row_ajax($form, $form_state) {
+  $commands = array();
+  // Display error messages in the form if any.
+  if (form_get_errors()) {
+    $html = '';
+    $html .= theme('status_messages');
+    $html .= backdrop_render($form);
+    $title = isset($form['#title']) ? $form['#title'] : backdrop_get_title();
+    $commands[] = ajax_command_open_modal_dialog($title, $html, array('dialogClass' => 'layout-dialog'));
+  }
+  else {
+    $commands[] = ajax_command_close_modal_dialog();
+
+    $update_ajax = layout_ajax_form_update($form, $form_state);
+    $commands = array_merge($commands, $update_ajax['#commands']);
+  }
+
+  return array(
+    '#type' => 'ajax',
+    '#commands' => $commands,
+  );
+}
+
+/**
+ * Menu callback to delete a row from a flexible template.
+ *
+ * @param $flexible_template
+ *   The loaded flexible template object.
+ * @param $original_row
+ *   The row above or below which a new row is being inserted.
+ *
+ * @ingroup forms
+ */
+function layout_flexible_template_delete_row($flexible_template, $original_row) {
+  // Todo use token, like in block add page.
+
+  $commands = array();
+  $flexible_template_id = $flexible_template['name'];
+
+  unset($flexible_template['rows'][$original_row]);
+  layout_set_flexible_tempstore($flexible_template);
+
+  if (backdrop_is_ajax()) {
+    $commands[] = ajax_command_remove('#flexible-row-id-' . $original_row);
+
+    return array(
+      '#type' => 'ajax',
+      '#commands' => $commands,
+    );
+  }
+  else {
+    backdrop_set_message(t('Row "@title" removed.', array('@title' => $flexible_template['title'])));
+    backdrop_goto('admin/structure/layouts/settings/flexible-template/' . $flexible_template_id . '/configure');
+  }
+}
+
+/**
+ * Menu callback; Delete a flexible template.
+ *
+ * @ingroup forms
+ */
+function layout_flexible_template_delete_form($form, &$form_state, $flexible_template) {
+  $form_state['template'] = $flexible_template;
+  $question = t('Delete flexible template @title?', array('@title' => $flexible_template['title']));
+  $description = t('This action cannot be undone.');;
+  backdrop_set_title($question, PASS_THROUGH);
+
+  // Check if this template is in use.
+  $errors = array();
+  foreach (layout_load_all() as $layout) {
+    if ($layout->layout_template == $flexible_template['name']) {
+      $errors[] = $layout->title;
+    }
+  }
+
+  if (!empty($errors)) {
+    $form['template_in_use'] = array(
+      '#title' => t('This template is in use in the following layouts') . ':',
+      '#type' => 'item',
+      '#markup' => theme('item_list', array('items' => $errors)),
+    );
+    backdrop_set_message(t('The "@layout" layout template is currently in use and may not be deleted.', array('@layout' => $flexible_template['title'])), 'error');
```

Comment:
Any reason why we are using both a form element + an error message here? I would ditch the form element, and include the list of layouts that are using the template within the error message. This is what we are doing elsewhere.

Also, instead of wrapping the layout name placeholder in double quotes, you should use `%layout` instead of the regular `@layout`; this would render it wrapped in `<em>`.